src/hipscat/io/file_io/file_io.py

```diff
@@ -0,0 +1,74 @@
+import json
+import os
+
+import pandas as pd
+
+from hipscat.io.file_io.file_pointers import FilePointer
+
+
+def make_directory(file_pointer: FilePointer, exist_ok: bool = False):
+    """Make a directory at a given file pointer
+
+    Will raise an error if a directory already exists, unless `exist_ok` is True in which case
+    any existing directories will be left unmodified
+
+    Args:
+        file_pointer: location in file system to make directory
+        exist_ok: Default False. If false will raise error if directory exists. It true existing
+        directories will be ignored and not modified
+
+    Raises:
+        OSError
+    """
+    os.makedirs(file_pointer, exist_ok=exist_ok)
+
+
+def write_string_to_file(
+    file_pointer: FilePointer, string: str, encoding: str = "utf-8"
+):
+    """Write a string to a text file
+
+    Args:
+        file_pointer: file location to write file to
+        string: string to write to file
+        encoding: Default: 'utf-8', encoding method to write to file with
+    """
+    with open(file_pointer, "w", encoding=encoding) as file:
+        file.write(string + "\n")
+
+
+def load_json_file(file_pointer: FilePointer, encoding: str = "utf-8") -> dict:
+    """Load a json file to a dictionary
+
+    Args:
+        file_pointer: location of file to read
+        encoding: string encoding method used by the file
+
+    """
+    json_dict = None
+    with open(file_pointer, "r", encoding=encoding) as json_file:
+        json_dict = json.load(json_file)
+    return json_dict
+
+
+def load_csv_to_pandas(file_pointer: FilePointer, **kwargs):
+    """Load a csv file to a pandas dataframe
+
+    Args:
+        file_pointer: location of csv file to load
+        **kwargs: arguments to pass to pandas `read_csv` loading method
+    """
+    return pd.read_csv(file_pointer, **kwargs)
```

Comment:
I'm not sure this load or write function below really need to be functions within file_io.py. It seems like it would be cleaner to just call `pd.read_csv(file_pointer)` directly from the various pieces of code. I might be missing something though.

Reply:
Yeah, I went with it this way in case we decided to go with some file representation in the future that didn't just work with pandas like that. Then we'd have to make a method to do extra stuff to get it to work anyway and it's easier to have the wrapper now than to refactor later. But that could be me being overly cautious and making more work for us.

Reply:
That's fair.